wxt_cidrbot/cidrbot.py

```diff
@@ -172,31 +172,70 @@ def webhook_request(self, event):
-            message_first_part = ""
-            remainder_message = ""
-            for repo in split_message:
-                self.logging.debug("Length of repo %s", len(repo))
-                if len(message_first_part) + len(repo) < self.WEBEX_MESSAGE_CHAR_LIMIT:
-                    if '**All Issues:**' not in repo:
-                        message_first_part += split_keyword
-                    message_first_part += repo
-                elif len(remainder_message) + len(repo) < self.WEBEX_MESSAGE_CHAR_LIMIT:
-                    remainder_message += "Repo:" + repo
-                else:
-                    remainder_message = "One repo has too many issues and exceeds the webex message limit."
+        split_message = message.split(split_keyword)
+        message_first_part = ""
+        msgs_to_be_sent = []
+
+        for repo in split_message:
+            self.logging.debug("Length of repo %s", len(repo))
+            if len(message_first_part) + len(repo) < self.WEBEX_MESSAGE_CHAR_LIMIT:
+                if '**All Issues:**' not in repo:
+                    message_first_part += split_keyword
+                message_first_part += repo
+                continue
+
+            split_repo = repo.split("\n")
+            repo_name = split_repo.pop(0)
+
+            #check to see if cannot append at least n lines to last message
+            is_not_able_to_append = len(message_first_part) + len(
+                "".join(split_repo[:num_needed_repo_lines])
+            ) >= self.WEBEX_MESSAGE_CHAR_LIMIT
+
+            #check see if entire repo fit in one message so don't waste memory and time splitting and repiecing
+            if is_not_able_to_append and len(repo) < self.WEBEX_MESSAGE_CHAR_LIMIT:
+                msgs_to_be_sent.append(message_first_part)
+                message_first_part += split_keyword + repo
+                continue
+            elif is_not_able_to_append:
+                msgs_to_be_sent.append(message_first_part)
+                message_first_part = ""
+
+            message_first_part += split_keyword + repo_name
+            for repo_part in split_repo:
+                if len(message_first_part) + len(repo_part) < self.WEBEX_MESSAGE_CHAR_LIMIT:
+                    message_first_part += "\n" + repo_part
+                    continue
+                msgs_to_be_sent.append(message_first_part)
+                #if end message, don't do split_keyword part
+                message_first_part = split_keyword + repo_name + " (continued)" + "\n" + repo_part
+        msgs_to_be_sent.append(message_first_part)
 
         if request_type == 'daily_message':
-            self.send_wbx_msg(room_id, message_first_part, pt_id)
-        if request_type == 'edit_message':
-            self.webex.edit_message(message_id, message_first_part, room_id)
-        self.send_wbx_msg(room_id, remainder_message, pt_id)
+            self.send_wbx_msg(room_id, msgs_to_be_sent.pop(0), pt_id)
```

Comment:
To reduce code complexity, you can split the function here, and delegate to another function to send messages and return the true